packages/expo-av/android/src/main/cpp/JAVManager.cpp

```diff
@@ -0,0 +1,126 @@
+
+#include <jni.h>
+#include <fbjni/fbjni.h>
+
+#include <memory>
+#include <string>
+
+#include <ReactCommon/CallInvokerHolder.h>
+#include <ReactCommon/CallInvoker.h>
+
+#include <android/log.h>
+
+namespace expo {
+
+using namespace facebook;
+using namespace jni;
+
+using TSelf = local_ref<HybridClass<expo::JAVManager>::jhybriddata>;
+
+TSelf JAVManager::initHybrid(alias_ref<jhybridobject> jThis) {
+    return makeCxxInstance(jThis);
+}
+
+void JAVManager::registerNatives() {
+    registerHybrid({
+        makeNativeMethod("initHybrid", JAVManager::initHybrid),
+        makeNativeMethod("installJSIBindings", JAVManager::installJSIBindings),
+    });
+}
+
+JPlayerData* JAVManager::getMediaPlayerById(int id) {
+    static const auto func = javaPart_->getClass()->getMethod<JPlayerData*(jint)>("getMediaPlayerById");
+    auto result = func(javaPart_.get(), id);
+    return result->cthis();
+}
+
+void JAVManager::installJSIBindings(jlong jsRuntimePointer,
+                                    jni::alias_ref<facebook::react::CallInvokerHolder::javaobject> jsCallInvokerHolder) {
+    auto& runtime = *reinterpret_cast<jsi::Runtime*>(jsRuntimePointer);
+    auto callInvoker = jsCallInvokerHolder->cthis()->getCallInvoker();
+
+    auto function = [this, callInvoker](jsi::Runtime &runtime,
+                                        const jsi::Value &thisValue,
+                                        const jsi::Value *args,
+                                        size_t argsCount) -> jsi::Value {
+        auto playerId = args[0].asNumber();
+
+        auto mediaPlayer = getMediaPlayerById(static_cast<int>(playerId));
+        if (mediaPlayer == nullptr) {
+            auto message = "Sound Instance with ID " + std::to_string(playerId) + "does not exist!";
+            throw jsi::JSError(runtime, message.c_str());
+        }
+
+        if (argsCount > 1 && args[1].isObject() && !args[1].isUndefined()) {
+            // second parameter received, it's the callback function.
+            auto message = "Setting Audio Sample Buffer Callback for Player " + std::to_string(playerId) + "...";
+            __android_log_write(ANDROID_LOG_INFO, TAG, message.c_str());
+
+            auto callback = args[1].asObject(runtime).asFunction(runtime);
+            auto callbackShared = std::make_shared<jsi::Function>(std::move(callback));
+
+            mediaPlayer->setSampleBufferCallback([callbackShared, &runtime, callInvoker](jni::alias_ref<jni::JArrayByte> sampleBuffer, double positionSeconds)  {
+                auto channelsCount = /* TODO: channelsCount */ 1;
+                auto size = sampleBuffer->size();
+
+                // TODO: Avoid copy by directly using ArrayBuffer? Or accessing values in the sampleBuffer?
+                // copies the JNI array into a vector and reinterprets it as an unsigned 8 bit int (u_byte)
+                std::vector<uint8_t> buffer(size);
+                sampleBuffer->getRegion(0, size, (int8_t*)buffer.data());
```

Comment:
which is it int or uint?
If you have a good reason for doing it this way change that cstyle cast to reinterpret cast and if not change type of the vector